storage/querybuilder/sql.go

```
@@ -0,0 +1,253 @@
+	// TODO: remove this comment or replace with my clear intention + a understandable language
+	// earlier time -----e-------------------c--------s----- newer time
+	// if start > end: descending
+	// and if cursor is present as well, id < c
+	// earlier time -----s------c---------------------e----- newer time
+	// else: ascending
+	// and if cursor is present: id > c
+
+	// cursor
+	if q.Cursor != "" {
+		if q.Start.After(q.End) {
+			queryParts = append(queryParts, `(id < ?)`)
```

golangci-lint (2.11.4)

[error] append to slice `queryParts` with non-zero initialized length

(makezero)